include/quicr/detail/uintvar.h

```diff
@@ -47,84 +49,66 @@ namespace quicr {
-     */
-    inline UintV ToUintV(uint64_t value)
-    {
-        constexpr uint64_t kLen1 = (static_cast<uint64_t>(-1) << (64 - 6) >> (64 - 6));
-        constexpr uint64_t kLen2 = (static_cast<uint64_t>(-1) << (64 - 14) >> (64 - 14));
-        constexpr uint64_t kLen4 = (static_cast<uint64_t>(-1) << (64 - 30) >> (64 - 30));
+        UintVar(Span<const uint8_t> bytes)
+          : _value{ 0 }
+        {
+            if (bytes.empty() || bytes.size() > sizeof(uint64_t) || bytes.size() != Size(bytes[0])) {
+                throw std::invalid_argument("Invalid bytes for uintvar");
+            }
 
-        if (value & (0xC0ull << 56)) { // Check if invalid
-            return {};
+            std::memcpy(&_value, bytes.data(), bytes.size());
         }
 
-        std::size_t length = sizeof(uint8_t);
-        if (value > kLen4) { // 62 bit encoding (8 bytes)
-            value |= (0xC0ull << 56);
-            length = sizeof(uint64_t);
-        } else if (value > kLen2) { // 30 bit encoding (4 bytes)
-            value |= (0x80u << 24);
-            length = sizeof(uint32_t);
-        } else if (value > kLen1) { // 14 bit encoding (2 bytes)
-            value |= (0x40u << 8);
-            length = sizeof(uint16_t);
+        operator uint64_t() const noexcept
+        {
+            return SwapBytes((_value & SwapBytes(uint64_t(~(~0x3Full << 56)))) << (sizeof(uint64_t) - Size()) * 8);
         }
 
-        value = SwapBytes(value);
-        const auto* bytes = reinterpret_cast<uint8_t*>(&value);
-        return { bytes + (sizeof(uint64_t) - length), bytes + sizeof(uint64_t) };
-    }
+        Span<const uint8_t> Bytes() const noexcept { return Span{ reinterpret_cast<const uint8_t*>(&_value), Size() }; }
 
-    /**
-     * @brief Convert Variable-Length Integer to uint64_t
-     *
-     * @param uintV Encoded variable-Length integer
-     *
-     * @returns uint64_t value of the variable length integer
-     */
-    inline uint64_t ToUint64(const UintV& uint_v)
-    {
-        if (uint_v.empty()) {
-            return 0;
+        static constexpr std::size_t Size(uint8_t msb_bytes) noexcept
+        {
+            if ((msb_bytes & 0xC0) == 0xC0) {
+                return sizeof(uint64_t);
+            } else if ((msb_bytes & 0x80) == 0x80) {
+                return sizeof(uint32_t);
+            } else if ((msb_bytes & 0x40) == 0x40) {
+                return sizeof(uint16_t);
+            }
+
+            return sizeof(uint8_t);
         }
 
-        uint64_t value{ 0 };
-        uint8_t* byte_value = reinterpret_cast<uint8_t*>(&value) + (sizeof(uint64_t) - uint_v.size());
-        std::memcpy(byte_value, uint_v.data(), uint_v.size());
-
-        byte_value[0] &= 0x3f; // Zero MSB length bits
+        constexpr std::size_t Size() const noexcept { return UintVar::Size(static_cast<uint8_t>(_value)); }
 
-        return SwapBytes(value);
-    }
+      private:
+        uint64_t _value;
```

Comment:
Based on the implementation, it's a bit misleading that type of uint64_t value is not the host order (unless big-endian host) value. It's actually used for storage of the var int, which is in network byte order.

I suggest a name change here to `bytes_` with a comment  such as "Variable length integer bytes storage"